tasks/code_manager_enabled.rb

```diff
@@ -0,0 +1,75 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+# frozen_string_literal: true
+
+require 'json'
+require 'uri'
+require 'net/http'
+require 'puppet'
+
+# GetPEAdmConfig task class
+class GetPEAdmConfig
+  def initialize(params); end
+
+  def execute!
+    code_manager_enabled = groups.dig('PE Master', 'classes', 'puppet_enterprise::profile::master', 'code_manager_auto_configure')
+    
+    puts({"code_manager_enabled" => code_manager_enabled}.to_json)
+  end
+
+  # Returns a GetPEAdmConfig::NodeGroups object created from the /groups object
+  # returned by the classifier
+  def groups
+    @groups ||= begin
+      net = https(4433)
+      res = net.get('/classifier-api/v1/groups')
+      NodeGroup.new(JSON.parse(res.body))
+    end
+  end
+
+  def https(port)
+    https = Net::HTTP.new('localhost', port)
+    https.use_ssl = true
+    https.cert = @cert ||= OpenSSL::X509::Certificate.new(File.read(Puppet.settings[:hostcert]))
+    https.key = @key ||= OpenSSL::PKey::RSA.new(File.read(Puppet.settings[:hostprivkey]))
+    https.verify_mode = OpenSSL::SSL::VERIFY_NONE
```

Comment:
I think VERIFY_NONE is quite unsafe. Is that required here, can't we switch to OpenSSL::SSL::VERIFY_PEER?

Reply:
Hey @bastelfreak, would like you to clarify this "unsafe" statement. How can this be unsafe on the current https method.

Reply:
because the TLS certificate isn't verified, this opens a door for potential men in the middle attacks.

Reply:
Current method is called from localhost to reach localhost. If MItM can possibly happen that would need someone to be already signed in localhost..... if someone is already on locahost, its already game over. What door opens from 127.0.0.1 to 127.0.0.1?

Reply:
The subtle problem is that you're talking to `localhost`, not `127.0.0.1`. So if someone manipulates DNS responses you open the door for man in the middle. Yes the risk is small, but I think the dangerous scenario is that people get used to `VERIFY_NONE` and then use it even in situations where the risk is way higher. And on the other side importing the cacert and verifying the chain isn't a real overhead.

Reply:
To manipulate the DNS responses of localhost you need to edit /etc/hosts on the local machine, here is the primary server. To do that, you need escalated permissions as user ... meaning that you are already in the machine, and It's too late and game over for security. Still, my question stands, what door opens when this script as a client makes a request to the local running server and doesn't check the certificate? Why this is a blocking issue? As for the people used to not verify certificates, cant see this as an example of that case, nor is a reason to block a PR for it... overhead or not, an example of how to import the cacert and verify the cert will be a better comment than Block and calling things unsafe.

Reply:
@bastelfreak Can you take a look now, please? Added verify peer, and changed from localhost to the cert of the host.

Reply:
@ragingra looks good, thanks a lot!

@CoMfUcIoS sorry, I was between incidents yesterday and had no time to provide examples, I usually try to do this. I know this sometimes looks a bit pedantic, but my experience is that those changes come back to bite us if we're not pedantic.